src/irust/cursor.rs

```diff
@@ -129,68 +137,81 @@ impl IRust {
-    pub fn goto_cursor(&mut self) -> Result<(), IRustError> {
-        self.cursor.goto(
-            self.internal_cursor.screen_pos.0 as u16,
-            self.internal_cursor.screen_pos.1 as u16,
-        )?;
-        Ok(())
+    pub fn is_at_line_end(&self, irust: &IRust) -> bool {
+        irust.buffer.is_empty()
+            || self.pos.buffer_pos == StringTools::chars_count(&irust.buffer)
     }
 
-    pub fn at_line_end(&self) -> bool {
-        self.buffer.is_empty()
-            || self.internal_cursor.buffer_pos == StringTools::chars_count(&self.buffer)
+    pub fn screen_height_overflow_by_str(&self, irust: &IRust, out: &str) -> usize {
+        let new_lines =
+            (StringTools::chars_count(out) + self.pos.screen_pos.0) / irust.size.0;
+
+        self.screen_height_overflow_by_new_lines(irust, new_lines)
     }
 
-    pub fn move_cursor_left(&mut self, move_type: Move) -> Result<(), IRustError> {
-        self.internal_cursor.move_left(move_type);
-        self.goto_cursor()?;
+    pub fn screen_height_overflow_by_new_lines(&self, irust: &IRust, new_lines: usize) -> usize {
+        // if corrected y  + new lines < self.size.1 there is no overflow so unwrap to 0
+        (new_lines + self.pos.screen_pos.1).saturating_sub(irust.size.1)
+    }
 
+    pub fn reset_position(&mut self) -> Result<(), IRustError> {
+        if let Some(copy) = self.copy.take() {
+            self.pos = *copy;
+            self.copy = Some(Box::new(self.pos.clone()));
+        }
+        self.internal_cursor.reset_position()?;
         Ok(())
     }
 
-    pub fn move_cursor_right(&mut self) -> Result<(), IRustError> {
-        self.internal_cursor.move_right();
-        self.goto_cursor()?;
+    pub fn goto_internal_pos(&mut self) -> Result<(), IRustError> {
+        self.internal_cursor.goto(
+            self.pos.screen_pos.0 as u16,
+            self.pos.screen_pos.1 as u16,
+        )?;
+        Ok(())
+    }
+
+    pub fn move_right(&mut self, count: u16) -> Result<(), IRustError> {
+        self.internal_cursor.move_right(count);
+        self.goto_internal_pos()?;
 
         Ok(())
     }
 
-    pub fn screen_height_overflow_by_str(&self, out: &str) -> usize {
-        let new_lines =
-            (StringTools::chars_count(out) + self.internal_cursor.screen_pos.0) / self.size.0;
+    pub fn move_left(&mut self, count: u16) -> Result<(), IRustError> {
+        self.internal_cursor.move_left(count);
+        self.goto_internal_pos()
+    }
 
-        self.screen_height_overflow_by_new_lines(new_lines)
+    // NOTE(smolck): Should we return a Result<T, E>?
+    pub fn hide(&self) {
+        self.internal_cursor.hide().unwrap();
     }
 
-    pub fn screen_height_overflow_by_new_lines(&self, new_lines: usize) -> usize {
-        // if corrected y  + new lines < self.size.1 there is no overflow so unwrap to 0
-        (new_lines + self.internal_cursor.screen_pos.1).saturating_sub(self.size.1)
+    // NOTE(smolck): Should we return a Result<T, E>?
```

Comment:
@sigmaSd Should we return a `Result<(), IRustError>` here either?

Reply:
Same as above